cmd/bucket-handlers.go

```diff
@@ -746,6 +746,18 @@ func (api objectAPIHandlers) HeadBucketHandler(w http.ResponseWriter, r *http.Re
 		return
 	}
 
+	if globalDNSConfig != nil {
```

Comment:
Let's try to make etcd call only when we are not able to find the bucket in the local object storage and etcd is enabled